apps/web/app/(dash)/menu.tsx

```diff
@@ -97,20 +95,21 @@ function Menu() {
 	);
 
 	const handleSubmit = async (content?: string, spaces?: number[]) => {
-		setDialogOpen(false);
+		if (loading) return;
 
+		setLoading(true);
```

Comment:
Instead of manually checking for loading, I think there's a react hook useFormStatus(), you can get the pending state from it and use that, it just removes the need of using setLoading at multiple places